packages/core/src/utils/parse-type-id.test.ts

```diff
@@ -0,0 +1,20 @@
+import test from 'ava';
+
+import { parseTypeId } from './parse-type-id';
+
+const fixtures = [
+  't_uint256',
+  't_mapping(t_uint256,t_uint256)',
+  't_mapping(unknown,t_uint256)',
+  't_array(t_uint256)3_storage',
+  't_mapping(t_uint256,t_array(t_bool)dyn_storage)',
+  't_mapping(t_uint256,t_mapping(t_string_memory_ptr,t_address))',
+  't_function_internal_nonpayable(t_uint256)returns()',
```

Comment:
I opened up an old artifact (compiled with solc 0.5 probably) and I see they use $_ and _$ instead of ( and ). Are we translating that somewhere else?
```
"typeIdentifier": "t_function_internal_nonpayable$__$returns$__$",
```

Reply:
Yes! [`decodeTypeIdentifier`](https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/a5fa60a691846080fd7ba84e9a5d6601d304902c/packages/core/src/storage.ts#L171)